lib/LLVMIRCodeGen/LLVMIRGen.cpp

```diff
@@ -1343,6 +1343,32 @@ void LLVMIRGen::generateLLVMIRForDataParallelInstr(
     break;
   }
 
+  case Kinded::Kind::ModuloInstKind: {
+
+    auto *MI = cast<ModuloInst>(I);
+    auto *dest = MI->getDest();
+    auto *src = MI->getSrc();
+    auto *destPtr = emitBufferAddress(builder, dest, kernel, bufferToArgNum);
+    auto *srcPtr = emitBufferAddress(builder, src, kernel, bufferToArgNum);
+    auto *divisor =
+        emitConst(builder, MI->getDivisor(), dest->getElementType());
+
+    llvm::Function *F = nullptr;
+    // Need _kernel suffix since these operations are implemented as
+    // "data-parallel" kernels in libjit.
+    if (MI->getSignFollowDivisor()) {
+      F = getFunction("element_modulo_kernel_sign_follow_divisor",
+                      dest->getElementType());
+    } else {
+      F = getFunction("element_modulo_kernel_no_sign_follow_divisor",
+                      dest->getElementType());
+    }
+    auto *stackedOpCall = createCall(builder, F, {loopCount, divisor, srcPtr});
+    auto *destAddr = builder.CreateGEP(builder.getInt64Ty(), destPtr, loopCount,
+                                       "buffer.element.addr");
+    builder.CreateStore(stackedOpCall, destAddr);
+  } break;
```

Comment:
Style nit: elsewhere in this file we put `break` inside the closing brace.

Reply:
Actually I did that, but utils/format.sh kept putting it up!
I'll move it down.

Reply:
Oh, weird.  Well, clang-format is gospel here, so I'm cool with its ineffable decisions :)